### PR TITLE
Miguelhar.plat 4955.initial commit

### PR DESCRIPTION
**Pull request by miguelhar**
Initial commit

### CODE REVIEW

**Review by fraenkel — APPROVED** [review]
it would be prudent to start get something wired from platform-apps so we can test/verify.

**Comment by fraenkel** [comment]
why are you adding more commits?

File: submodules/k8s/main.tf [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  k8s_pre_setup_sh_filename     = "k8s-pre-setup.sh"
+  k8s_pre_setup_sh_template     = "k8s-pre-setup.sh.tftpl"
+  aws_auth_filename             = "aws-auth.yaml"
+  aws_auth_template             = "aws-auth.yaml.tftpl"
+  calico = {
+    operator_url         = "https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/${var.calico_version}/config/master/calico-operator.yaml"
+    custom_resources_url = "https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/${var.calico_version}/config/master/calico-crs.yaml"
+  }
+
+  k8s_tunnel_command = "docker run --rm --name ${local.mallory_container_name} -d -v $PWD/${local.mallory_config_filename}:${local.mallory_config_path_container} -p ${var.mallory_local_normal_port}:${var.mallory_local_normal_port} -p ${var.mallory_local_smart_port}:${var.mallory_local_smart_port} -v ${var.ssh_pvt_key_name}:${local.pvt_key_path_container} zoobab/mallory"
```
[the remark this answers is not in the document]
Comment (Secretions) [reply submodules/k8s/main.tf]: Sorry for not seeing this response two weeks ago, but I think I should rephrase my concern. Outside of platform-apps development environments, the deployer is always ran from within a docker container. So to work from the deployer, we can't be running mallory via docker container. We either need to have some mechanism for running mallory as a binary, or passing off the script to the deployer to run via its own proxy mode. Won't hold up getting the initial commit through, but it'll be a blocker for deployer integration.

**Review by Secretions — APPROVED** [review]
I'd rather get this committed and do anything else in followup, but I do agree with Michael that we should get this integrated into the deployer next so we capture all the things that need to be done for that (which the mallory thing I called out will probably be one of).

Also, tried it myself and got this:

```
odule.domino_eks.module.k8s_setup.null_resource.run_k8s_pre_setup (local-exec): Error response from daemon: Cannot kill container: mallory_k8s_tunnel: No such container: mallory_k8s_tunnel
╷
│ Error: local-exec provisioner error
│
│   with module.domino_eks.module.k8s_setup.null_resource.run_k8s_pre_setup,
│   on ../../terraform-aws-eks/submodules/k8s/main.tf line 83, in resource "null_resource" "run_k8s_pre_setup":
│   83:   provisioner "local-exec" {
│
│ Error running command 'k8s-pre-setup.sh': exit status 1. Output: + RED='\e[31m'
│ + GREEN='\e[32m'
│ + EC='\e[0m'
│ + source k8s-functions.sh
│ ++ RED='\e[31m'
│ ++ GREEN='\e[32m'
│ ++ EC='\e[0m'
│ + trap close_ssh_tunnel_to_k8s_api EXIT
│ + main
│ + open_ssh_tunnel_to_k8s_api
│ + printf '\e[32m Setting up mallory k8s tunnel... \e[0m \n'
│  Setting up mallory k8s tunnel...
│ + [[ -z '' ]]
│ + MALLORY_PORT=1316
│ + https_proxy=http://localhost:1316
│ + export https_proxy
│ + test -f mallory.json
│ + docker kill mallory_k8s_tunnel
│ Error response from daemon: Cannot kill container: mallory_k8s_tunnel: No such container: mallory_k8s_tunnel
│ + true
│ + eval docker run --rm --name mallory_k8s_tunnel -d -v /home/mrgus/domino/terraform-aws-eks/example/mallory.json:/root/.config/mallory.json -p 1315:1315 -p
│ 1316:1316 -v /home/mrgus/domino/terraform-aws-eks/example/home/mrgus/.ssh/domino-test.pem:/root/domino-test.pem zoobab/mallory
│ ++ docker run --rm --name mallory_k8s_tunnel -d -v /home/mrgus/domino/terraform-aws-eks/example/mallory.json:/root/.config/mallory.json -p 1315:1315 -p
│ 1316:1316 -v /home/mrgus/domino/terraform-aws-eks/example/home/mrgus/.ssh/domino-test.pem:/root/domino-test.pem zoobab/mallory
│ Unable to find image 'zoobab/mallory:latest' locally
│ latest: Pulling from zoobab/mallory
│ 54f7e8ac135a: Pulling fs layer
│ d6341e30912f: Pulling fs layer
│ 087a57faf949: Pulling fs layer
│ 5d71636fb824: Pulling fs layer
│ f368dba6a331: Pulling fs layer
│ 69ffff7c9ff9: Pulling fs layer
│ 2613ffabce6c: Pulling fs layer
│ b2c9d8b9c9a7: Pulling fs layer
│ 4f10ea82eadb: Pulling fs layer
│ 766af0e8f649: Pulling fs layer
│ 6b6c4662c258: Pulling fs layer
│ 4f10ea82eadb: Waiting
│ 69ffff7c9ff9: Waiting
│ b2c9d8b9c9a7: Waiting
│ 766af0e8f649: Waiting
│ 5d71636fb824: Waiting
│ 2613ffabce6c: Waiting
│ 087a57faf949: Verifying Checksum
│ 087a57faf949: Download complete
│ d6341e30912f: Verifying Checksum
│ d6341e30912f: Download complete
│ 54f7e8ac135a: Verifying Checksum
│ 54f7e8ac135a: Download complete
│ 5d71636fb824: Verifying Checksum
│ 5d71636fb824: Download complete
│ 54f7e8ac135a: Pull complete
│ d6341e30912f: Pull complete
│ 087a57faf949: Pull complete
│ 2613ffabce6c: Verifying Checksum
│ 2613ffabce6c: Download complete
│ 5d71636fb824: Pull complete
│ b2c9d8b9c9a7: Download complete
│ f368dba6a331: Verifying Checksum
│ f368dba6a331: Download complete
│ f368dba6a331: Pull complete
│ 69ffff7c9ff9: Verifying Checksum
│ 69ffff7c9ff9: Download complete
│ 766af0e8f649: Verifying Checksum
│ 766af0e8f649: Download complete
│ 69ffff7c9ff9: Pull complete
│ 2613ffabce6c: Pull complete
│ b2c9d8b9c9a7: Pull complete
│ 6b6c4662c258: Verifying Checksum
│ 6b6c4662c258: Download complete
│ 4f10ea82eadb: Verifying Checksum
│ 4f10ea82eadb: Download complete
│ 4f10ea82eadb: Pull complete
│ 766af0e8f649: Pull complete
│ 6b6c4662c258: Pull complete
│ Digest: sha256:0f047e0a5e266ccf15a7da4342a29c5792d99fc0066138e32c745dc7ec33e6e0
│ Status: Downloaded newer image for zoobab/mallory:latest
│ a3be6bcc9fd6307d661bb7ef58b4cc2d44061e0684fd2131848a6f7b1f51f054
│ + sleep 3
│ + echo
│
│ + check_kubeconfig
│ + KUBECONFIG=kubeconfig
│ + printf '\e[32m Checking if kubeconfig exists... \e[0m \n'
│  Checking if kubeconfig exists...
│ + test -f kubeconfig
│ + echo 'kubeconfig exists.'
│ kubeconfig exists.
│ + export KUBECONFIG
│ + echo
│
│ + set_k8s_auth
│ + AWS_AUTH_YAML=aws-auth.yaml
│ + test -f aws-auth.yaml
│ + printf '\e[32m Updating aws-auth.yaml... \e[0m \n'
│  Updating aws-auth.yaml...
│ + kubectl apply -f aws-auth.yaml
│ The connection to the server 6C5C306823552B64CF1425E3E4B37F07.sk1.us-west-2.eks.amazonaws.com was refused - did you specify the right host or port?
│ + close_ssh_tunnel_to_k8s_api
│ + printf '\e[32m Shutting down mallory k8s tunnel mallory_k8s_tunnel ... \e[0m'
│  Shutting down mallory k8s tunnel mallory_k8s_tunnel ... + docker kill mallory_k8s_tunnel
│ Error response from daemon: Cannot kill container: mallory_k8s_tunnel: No such container: mallory_k8s_tunnel
│
╵
╷
│ Error: error creating Flow Log (vpc-06ff22fd92181a64b): 1 error occurred:
│ 	* vpc-06ff22fd92181a64b: 400: Access Denied for LogDestination: ashtest1000-backups. Please check LogDestination permission
│
│
│
│   with module.domino_eks.module.network.aws_flow_log.this,
│   on ../../terraform-aws-eks/submodules/network/main.tf line 82, in resource "aws_flow_log" "this":
│   82: resource "aws_flow_log" "this" {
```

**Pull request merged**